src/relay/transforms/partition_graph.cc

```diff
@@ -148,25 +148,42 @@ class Partitioner : public ExprMutator {
       CHECK_EQ(call->args.size(), 1U);
 
       // Traverse the rest graph.
-      auto input_expr = VisitExpr(call->args[0]);
+      Expr parent = call->args[0];
+      auto input_expr = VisitExpr(parent);
+
+      // Backtrace the parent to find the LCA node that is not a begin/ end op
```

Comment:
Is it really LCA? To me it seems just an ancestor node.

Reply:
yeah, it should be LCA, like A-->end1->begin->B
                                                 -->end2->begin->C

If B and C are in the same region, A should have only one out this region instead of two. A is the LCA of B and C.